raiden/network/proxies/token_network.py

```diff
@@ -1120,6 +1138,23 @@ def _set_total_deposit(
                 Address(self.address), failed_at_blocknumber
             )
 
+            # This check can only be done against the block
```

Comment:
I am not sure I 100% understand what you are trying to say here. Can you try to explain it in a bit better words to me here so that we find the right contents for the comment to make sense?

So we can check if the channel id is invalid only when the channel is not settled? IF it's settled we seem to skip the check due to the `and` clause. Right?

If so we can even skip this completely by moving the channel state checks from below: https://github.com/raiden-network/raiden/pull/4779/files#diff-d02574a7a7eca826412780e80a6326a3R1158

above this check. Then the only valid states at the time of this check will be OPEN/CLOSED. And then you can simplify the comment to something like:

"This check can only be done if the channel is in the open/closed states because from the settled state and after the id is removed from the smart contracts."

Reply:
> I am not sure I 100% understand what you are trying to say here. Can you try to explain it in a bit better words to me here so that we find the right contents for the comment to make sense?

The smart contract deletes the data. If the channel is in any state other than `OPENED` or `CLOSED` there is no id on-chain to compare against anymore. 

> So we can check if the channel id is invalid only when the channel is not settled? IF it's settled we seem to skip the check due to the and clause. Right?

yes

I updated the comment with your suggestion.